namesys/routing.go

```diff
@@ -30,15 +30,29 @@ func NewRoutingResolver(route routing.IpfsRouting) Resolver {
 	return &routingResolver{routing: route}
 }
 
-// CanResolve implements Resolver. Checks whether name is a b58 encoded string.
-func (r *routingResolver) CanResolve(name string) bool {
+// newRoutingResolver returns a resolver instead of a Resolver.
+func newRoutingResolver(route routing.IpfsRouting) resolver {
+	if route == nil {
+		panic("attempt to create resolver with nil routing system")
+	}
+
+	return &routingResolver{routing: route}
+}
+
+// canResolve implements resolver. Checks whether name is a b58 encoded string.
+func (r *routingResolver) canResolve(name string) bool {
 	_, err := mh.FromB58String(name)
 	return err == nil
 }
 
-// Resolve implements Resolver. Uses the IPFS routing system to resolve SFS-like
-// names.
-func (r *routingResolver) Resolve(ctx context.Context, name string) (path.Path, error) {
+// Resolve implements Resolver.
+func (r *routingResolver) Resolve(ctx context.Context, name string, depth int) (path.Path, error) {
```

Comment:
On Thu, May 14, 2015 at 09:32:03PM -0700, Juan Batiz-Benet wrote:
> > -func (r *routingResolver) Resolve(ctx context.Context, name string) (path.Path, error) {
> > +// Resolve implements Resolver.
> > +func (r *routingResolver) Resolve(ctx context.Context, name string, depth int) (path.Path, error) {
> 
> i dont think we should encourage users to chose different resolution
> depths. it may be useful in some odd cases, but i want to minimize
> the variability of this as it is an important protocol concern.

So the only cases you want to expose are “resolve a single step” and
“resolve as many steps as you can without violating some configured
limit”?  I'm ok with that, but I'd like a clearer picture of how that
configured limit gets set and managed before taking this knob off.